lib/HttpMp4Video.tsx

```diff
@@ -152,13 +133,18 @@ export const HttpMp4Video: React.FC<HttpMp4VideoProps> = ({
     const videoEl = videoRef.current
 
     if (src !== undefined && src.length > 0 && videoEl !== null) {
+      const endedCallback = () => {
+        __onEndedRef.current?.()
+      }
       debugLog('create pipeline', src)
       const newPipeline = new pipelines.HttpMsePipeline({
         http: { uri: src },
         mediaElement: videoEl,
       })
       setPipeline(newPipeline)
 
+      newPipeline.onServerClose = endedCallback
```

Comment:
Is there a specific reason to not create `endedCallback` here on the fly?

Reply:
Nope, fixed that.